rllib/evaluation/rollout_worker.py

```diff
@@ -1944,15 +1989,41 @@ def _determine_spaces_for_multi_agent_dict(
                     "`observation_space` specified in config!"
                 )
 
-            multi_agent_dict[pid] = multi_agent_dict[pid]._replace(
+            multi_agent_policies_dict[pid] = multi_agent_policies_dict[pid]._replace(
                 observation_space=obs_space
             )
 
         if policy_spec.action_space is None:
             if spaces is not None and pid in spaces:
                 act_space = spaces[pid][1]
             elif env_act_space is not None:
-                act_space = env_act_space
+                # Multi-agent case AND different agents have different spaces:
+                # Need to reverse map spaces (for the different agents) to certain
+                # policy IDs.
+                if isinstance(env, MultiAgentEnv) and env._spaces_in_preferred_format:
+                    act_space = None
+                    mapping_fn = policy_config.get("multiagent", {})\
+                        .get("policy_mapping_fn", None)
+                    if mapping_fn:
+                        for aid in env.get_agent_ids():
+                            # Match: Assign spaces for this agentID to the policy ID.
+                            if mapping_fn(aid, None, None) == pid:
+                                # Make sure, different agents that map to the same
+                                # policy don't have different spaces.
+                                if (
+                                    act_space is not None
+                                    and env_act_space[aid] != act_space
+                                ):
+                                    raise ValueError(
+                                        "Two agents in your environment map to the same"
+                                        " policyID (as per your `policy_mapping_fn`), "
```

Comment:
is this a common issue?

Reply:
Not sure, but I wanted to cover this error here.

Possible that a users does this wrong, right? Defining two separate action spaces for two agents in the env, but then setting in their config to be mapped to the same policy.